### ISSUE TITLE
Starting Java 17 the security manager will be broken

### ISSUE THREAD

**Issue by tjwatson** [osgi/osgi#223]
https://openjdk.java.net/jeps/411

See https://docs.osgi.org/javadoc/osgi.core/8.0.0/org/osgi/framework/Constants.html#FRAMEWORK_SECURITY
See https://docs.osgi.org/javadoc/osgi.core/8.0.0/org/osgi/framework/Constants.html#FRAMEWORK_SECURITY_OSGI

If framework launch property is set `org.osgi.framework.security=osgi` then the framework is obligated to set the security manager.  But starting with Java 17 (according to https://openjdk.java.net/jeps/411) the JVM will start throwing UnsupportedOperationException if the framework dynamically sets the security manager unless the property `java.security.manager=allow` is set when Java is launched.

Should the framework fail to launch with an UnsupportedOperationException when this happens?  This means all existing launch configurations for the framework will cease to launch until their own JVM launch properties are updated to set `-Djava.security.manager=allow` or should the framework avoid propagating the UnsupportedOperationException and allow the framework to launch without a security manager set in this case?

**Comment by bjhargrave** [comment]
I think that if the user has requested a SecurityManager via the framework launch property being set to `org.osgi.framework.security=osgi`, then a framework implementation must install a SecurityManager. If the JRE does not allow it, the framework impl must report that. An up-to-date framework impl could know about the Java 17 changes and, in the error message of the exception, advise the user of the `-Djava.security.manager=allow` JRE invocation option.

I do not think a framework implementation should proceed without a SecurityManager when once was requested. That seems a security vulnerability.

**Comment by bjhargrave** [comment]
But we probably need to start long term thinking about deprecation/removal of all Security sections in the OSGi specs if Java permissions are indeed slated for the chopping block.

**Comment by rickhall** [comment]
Agree with BJ, you can't start without a security manager.

**Comment by tjwatson** [comment]
I also agree.  Leaving open in case we want to make some explicit statement in the spec.  May want to decide if the UnsupportedOperationException is propagated or if Framework.init just throws a BundleException with the cause set.

**Comment by laeubi** [comment]
Starting with Java 24 the Security Manager is **permanently disabled**

https://lists.apache.org/thread/jfoz60p7l23gt77dvoyzp367h0g24j0n

this already has impacts on Equinox with Java 21:
- https://github.com/eclipse-equinox/equinox/pull/728

where it shows that for some reason the CM TCKs no longer run:
- https://github.com/eclipse-equinox/equinox/pull/728/checks?check_run_id=34634358950

As a short term solution the TCKs needs to be adjusted to have an `asumeThat(...)` or similar that make sure security manager can actually be used and skip otherwise.
